internal/services/nginx/nginx_deployment_resource.go

```diff
@@ -435,7 +438,12 @@ func (m DeploymentResource) Create() sdk.ResourceFunc {
 				prop.NetworkProfile.NetworkInterfaceConfiguration.SubnetId = pointer.FromString(model.NetworkInterface[0].SubnetId)
 			}
 
-			if model.Capacity > 0 {
+			isBasicSKU := strings.HasPrefix(model.Sku, "basic")
+			if isBasicSKU && (model.Capacity != defaultCapacity || len(model.AutoScaleProfile) > 0) {
+				return fmt.Errorf("basic SKUs are incompatible with capacity or auto_scale_profiles")
```

Comment:
Minor, but we put property names between back ticks in error messages.
```suggestion
				return fmt.Errorf("basic SKUs are incompatible with `capacity` or `auto_scale_profiles`")
```